templates/cc/repeated.go

```diff
@@ -24,15 +24,23 @@ const repTpl = `
 	{{ end }}
 
 	{{ if $r.GetUnique }}
-	std::set<std::reference_wrapper<const {{ $typ }}>> {{ lookup $f "Unique" }};
+	// Implement comparison for wrapped reference types
+	struct cmp {
+		bool operator() (const std::reference_wrapper<{{ $typ }}> lhs, const std::reference_wrapper<{{ $typ }}> rhs) const {
+			return lhs.get() < rhs.get();
+		}
+	};
+
+	// Save a set of references to avoid copying overhead
+	std::set<std::reference_wrapper<{{ $typ }}>,cmp> {{ lookup $f "Unique" }};
```

Comment:
Can you elaborate on the choice of reference wrapper here? For wrapper/string/bytes types, you can just use a reference type to instantiate the template, you don't save anything major by not copying a uint32/uint64 for the primitive types.

Reply:
My reading seems to suggest that you can't just use a reference type, and that using std::reference_wrapper is the "right" way to build an STL collection of references: https://stackoverflow.com/questions/7707842/stl-containers-with-reference-to-objects

Reply:
Oh right, brain fart, you are right.

Reply:
LGTM modulo the question of set ordering.